plugins/ufm_log_analyzer_plugin/src/loganalyze/logs_extraction/tar_extractor.py

```diff
@@ -125,20 +128,37 @@ def extract_files(
             ]
             for inner_tar_name in inner_tar_files:
                 with outer_tar.extractfile(inner_tar_name) as inner_tar_stream:
+
+                    # Check if the inner stream can be read
+                    try:
+                        # Read some bytes to verify the file is not corrupted
+                        inner_tar_stream.peek(1)  # Peek at the first byte
+                    except Exception as e:
+                        log.Logger.info("Error reading inner tar file %s: %s", inner_tar_name, e)
+                        continue  # Skip this file if it's invalid
```

Comment:
Why do we need this validation if we also added the try-catch for the specific Exception in the next lines?

Reply:
You’re right, that was my confusion. It’s fixed.